core/includes/common.inc

```diff
@@ -483,18 +483,18 @@ function backdrop_get_feeds($delimiter = "\n") {
 /**
  * Processes a URL query parameter array to remove unwanted elements.
  *
- * @param $query
+ * @param ?array|null $query
```

Comment:
In order to make it easier to work with PHPStan, not harder, we should use a comment syntax it recognizes (and doesn't flag as error).

I'd suggest to stick with `@param array|null` (no question mark for nullable types in doc block).

I didn't find any official recommendation for notation of nullable types in doc blocks, though.

Reply:
You're right, I thought PHPStorm (which I believe uses PHPStan) only worked with `?array` syntax, but it looks like `array|null` is also totally acceptable. Doing `?array|null` is redundant, though in PHPStorm it does not throw any linting errors if both are used. If `array|null` throws no errors anywhere, let's stick with that as you suggest.

Reply:
PR is updated to use `@param array|null` syntax everywhere, which makes for fewer overall changes and is in my opinion, easier to understand.